examples/js/loaders/GLTFLoader.js

```diff
@@ -2132,184 +2224,220 @@ THREE.GLTFLoader = ( function () {
 
-					var material = primitive.material === undefined
-						? createDefaultMaterial()
-						: dependencies.materials[ primitive.material ];
+					var geometry = geometries[ i ];
+					var primitive = primitives[ i ];
 
-					if ( material.aoMap
-							&& geometry.attributes.uv2 === undefined
-							&& geometry.attributes.uv !== undefined ) {
+					// 1. create Mesh
 
-						console.log( 'THREE.GLTFLoader: Duplicating UVs to support aoMap.' );
-						geometry.addAttribute( 'uv2', new THREE.BufferAttribute( geometry.attributes.uv.array, 2 ) );
+					var mesh;
 
-					}
+					var material = isMultiMaterial ? originalMaterials : originalMaterials[ i ]
 
-					// If the material will be modified later on, clone it now.
-					var useVertexColors = geometry.attributes.color !== undefined;
-					var useFlatShading = geometry.attributes.normal === undefined;
-					var useSkinning = meshDef.isSkinnedMesh === true;
-					var useMorphTargets = primitive.targets !== undefined;
+					if ( primitive.mode === WEBGL_CONSTANTS.TRIANGLES ||
+						primitive.mode === WEBGL_CONSTANTS.TRIANGLE_STRIP ||
+						primitive.mode === WEBGL_CONSTANTS.TRIANGLE_FAN ||
+						primitive.mode === undefined ) {
 
-					if ( useVertexColors || useFlatShading || useSkinning || useMorphTargets ) {
+						// .isSkinnedMesh isn't in glTF spec. See .markDefs()
+						mesh = meshDef.isSkinnedMesh === true
+							? new THREE.SkinnedMesh( geometry, material )
+							: new THREE.Mesh( geometry, material );
 
-						if ( material.isGLTFSpecularGlossinessMaterial ) {
+						if ( primitive.mode === WEBGL_CONSTANTS.TRIANGLE_STRIP ) {
 
-							var specGlossExtension = extensions[ EXTENSIONS.KHR_MATERIALS_PBR_SPECULAR_GLOSSINESS ];
-							material = specGlossExtension.cloneMaterial( material );
+							mesh.drawMode = THREE.TriangleStripDrawMode;
 
-						} else {
+						} else if ( primitive.mode === WEBGL_CONSTANTS.TRIANGLE_FAN ) {
 
-							material = material.clone();
+							mesh.drawMode = THREE.TriangleFanDrawMode;
 
 						}
 
-					}
+					} else if ( primitive.mode === WEBGL_CONSTANTS.LINES ) {
+
+						mesh = new THREE.LineSegments( geometry, material );
+
+					} else if ( primitive.mode === WEBGL_CONSTANTS.LINE_STRIP ) {
 
-					if ( useVertexColors ) {
+						mesh = new THREE.Line( geometry, material );
 
-						material.vertexColors = THREE.VertexColors;
-						material.needsUpdate = true;
+					} else if ( primitive.mode === WEBGL_CONSTANTS.LINE_LOOP ) {
+
+						mesh = new THREE.LineLoop( geometry, material );
+
+					} else if ( primitive.mode === WEBGL_CONSTANTS.POINTS ) {
+
+						mesh = new THREE.Points( geometry, material );
+
+					} else {
+
+						throw new Error( 'THREE.GLTFLoader: Primitive mode unsupported: ' + primitive.mode );
 
 					}
 
-					if ( useFlatShading ) {
+					if ( primitive.targets !== undefined ) {
 
-						material.flatShading = true;
+						addMorphTargets( mesh, meshDef, primitive, dependencies.accessors );
```

Comment:
In case that we merge geometries with THREE.BufferGeometryUtils, we need to combine morphPosition/Normal accessors somewhere...

Reply:
Hm yeah.. note that [BufferGeometryUtils can do that](https://github.com/mrdoob/three.js/blob/master/examples/js/BufferGeometryUtils.js#L295-L324) if the morphAttributes are set when it's called.

Reply:
Yup I'm aware of that, too and trying to set morphAttributes in `loadGeometries()` before combining.

Reply:
Fixed.